### PR TITLE
tests, utils: Extract container lookup helpers to libpod

### PR DESCRIPTION
**Pull request by EdDev**
### What this PR does

Extract the helpers out of utils and rename them to fit nicely in their new home package.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/approve

File: tests/libpod/container.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+func LookupComputeContainer(pod *v1.Pod) *v1.Container {
+	return LookupContainer(pod, computeContainer)
+}
+
+func LookupContainer(pod *v1.Pod, containerName string) *v1.Container {
+	for _, container := range pod.Spec.Containers {
+		if container.Name == containerName {
+			return &container
```
Comment (orelmisan) [inline tests/libpod/container.go:39]: Could there be an issue caused by returning the address of the iteration variable?

In comparison to:
```go
for i := range pod.Spec.Containers {
	if pod.Spec.Containers[i].Name == containerName {
		return &pod.Spec.Containers[i]
	}
}
```

File: tests/libpod/container.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+package libpod
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	computeContainer = "compute"
```
Comment (orelmisan) [inline tests/libpod/container.go:29]: nit:
```suggestion
	computeContainerName = "compute"
```

**Review by orelmisan — COMMENTED** [review]
Thank you for the cleanup @EdDev 
Just one question inline.

File: tests/libpod/container.go (hunk 2 above)
Comment (EdDev) [reply tests/libpod/container.go:29]: Looks like a nit to me, but if you insist I will push again.

File: tests/libpod/container.go (hunk 1 above)
Comment (EdDev) [reply tests/libpod/container.go:39]: Yes, but it will take time until we can depend on it.

**Comment by EdDev** [comment]
change: Answered review comments.

**Review by orelmisan — APPROVED** [review]
Thank you

**Pull request merged**